### ISSUE TITLE
Docker deployment on Windows

### ISSUE THREAD

**Issue by mausch** [DataDog/docker-dd-agent#175]
Now that Windows supports Docker and [Kubernetes support is under development](https://kubernetes.io/docs/getting-started-guides/windows/), has anyone looked into the possibility of deploying the Datadog agent on Windows through Docker?

For starters, this would obviously mean a separate Dockerfile. One thing I'm not sure about is how the agent picks up resources from the OS on Windows and if it would be possible to do that in a container. I think this should be possible using the [`process` isolation](https://docs.docker.com/engine/reference/commandline/run/#/specify-isolation-technology-for-container---isolation) but tbh I've not used Windows containers much.

Anyway this is just to start the discussion. It might not even be viable at all.

**Comment by IvanBoyko** [comment]
Voting up!
Especially there was ability to monitor via TCP socket (env var DOCKER_HOST), not named pipe (Windows way of Unix socket)

**Comment by davidobrien1985** [comment]
Is this something that's being worked on?
I'm using Azure Container Services with Windows agents and it's really a shame that I can't monitor them with datadog.

**Comment by hkaj** [comment]
@davidobrien1985 this is not being worked on currently, but we plan on supporting windows containers eventually. We are keeping track of feature requests and will tackle it at some point.

**Comment by guitarrapc** [comment]
I'm using AWS ECS with Windows Server container and it feel very nice if I can monitor with Datadog as like linux do.
Any plan is appreciated and if there are any stopping point, please share to community.

**Comment by tommymei** [comment]
Same here, running Windows Server containers and would like to have the datadog agent.

**Comment by Kralizek** [comment]
@hkaj any information about this?

**Comment by hkaj** [comment]
@Kralizek & others, if you already have Datadog account, please file a feature request for this with your CSM or in a support ticket.
As for updates, all I can say is that interest is still low (few requests, and for small numbers of servers) compared to other features, so we are not yet prioritizing it, although it's still on our radar.

**Comment by Kralizek** [comment]
> As for updates, all I can say is that interest is still low (few requests, and for small numbers of servers) compared to other features, so we are not yet prioritizing it, although it's still on our radar.

I would argue that the number of Windows Server servers running containers and running the Datadog agents is a small part of the number of servers running containers on Windows (think of PaaS offerings like AWS ECS or Azure Service Fabric)

**Comment by hkaj** [comment]
Agreed, what I meant is that users who request this feature only run a small number of Windows servers compared to Linux servers. It certainly does come up, but rarely, and as a low-priority request compared to other requests, so for now we have prioritize other features. When this gets bumped up in the list and we get an ETA we'll update this issue with new info.
